src/js/stores/MesosStateStore.js

```diff
@@ -90,79 +49,51 @@ class MesosStateStore extends GetSetBaseStore {
       listenAlways: true
     });
 
-    this.dispatcherIndex = AppDispatcher.register(payload => {
-      if (payload.source !== ActionTypes.SERVER_ACTION) {
-        return false;
-      }
-
-      var action = payload.action;
-      switch (action.type) {
-        case ActionTypes.REQUEST_MESOS_STATE_SUCCESS:
-          this.processStateSuccess(action.data);
-          break;
-        case ActionTypes.REQUEST_MESOS_STATE_ERROR:
-          this.processStateError(action.xhr);
-          break;
-        case ActionTypes.REQUEST_MESOS_STATE_ONGOING:
-          this.processOngoingRequest();
-          break;
-      }
-
-      return true;
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
     });
-
-    VisibilityStore.addChangeListener(
-      VISIBILITY_CHANGE,
-      this.onVisibilityStoreChange.bind(this)
-    );
   }
 
   addChangeListener(eventName, callback) {
     this.on(eventName, callback);
 
-    if (this.shouldPoll()) {
-      startPolling();
+    if (this.shouldSubscribe()) {
+      this.subscribe();
     }
   }
 
   removeChangeListener(eventName, callback) {
     this.removeListener(eventName, callback);
 
-    if (!this.shouldPoll()) {
-      stopPolling();
+    if (!this.shouldSubscribe()) {
+      this.unsubscribe();
     }
   }
 
-  onVisibilityStoreChange() {
-    if (!VisibilityStore.isInactive() && this.shouldPoll()) {
-      startPolling();
-
-      return;
-    }
-
-    stopPolling();
+  shouldSubscribe() {
```

Comment:
@nLight have we tested what happens if the tab is inactive for long periods of time and then refocussed? The VisibilityStore was introduced because we learned many users would keep the UI open but navigate to other tabs and other applications, then return to find the tab frozen. cc/ @orlandohohmeier

Reply:
We can test that, I decided since we're not polling anything anymore we don't need the Visibility check, but maybe we still do.